### ISSUE TITLE
Legend error unit.arithmetic() argument e2 is missing

### ISSUE THREAD

**Issue by jmw86069** [jokergoo/ComplexHeatmap#722]
I encountered a new error that was not there previously, I'm using ComplexHeatmap version 2.7.8.1000, and R-3.6.1. I think the error might be from using R-3.6.1 (and from grid-3.6.1). It appears only when using horizontal color legend, so most of my heatmaps are fine - but some heatmaps that worked previously are now showing an error.

I confirmed a potential fix with two changes described below. You can update how you feel is best. :) These changes confirm the source of error, with one possible workaround.

Reproducible example:
```R
col_fun <- circlize::colorRamp2(
   breaks=-2:2,
   colors=c("navy", "dodgerblue", "white", "tomato", "red3"))
lgd <- ComplexHeatmap::Legend(col_fun=col_fun,
   title="log2FC",
   direction="horizontal")
```

Output:
```
 Error in unit.arithmetic(.Generic, e1, e2) : 
  argument "e2" is missing, with no default 
```

The error `argument "e2" is missing, with no default` is caused by the expression `-ext` in `horizontal_continuous_legend_body()`. Changing to use `-1 * ext` fixes this error (for me).

After that fix, another error occurs when using a legend title.
(This example only shows the error after making the first fix above.)
```R
lgd <- ComplexHeatmap::Legend(col_fun=col_fun,
   title="log2FC",
   direction="horizontal")
```
Output:
```
 Error in Ops.unit(title_width, legend_width) : 
  operator '>' not meaningful for units 
```

This occurs in `Legend()` at this line:
```R
if(title_width > legend_width && title_position == "topleft") {
```
The comparison `title_width > legend_width` causes the error. This might be different in grid-3.6.1 compared to grid-4.0.0, I don't know?

Anyway, a *really simple* workaround that would help me tremendously, and which does not require any functional changes to the R code, is to reverse the conditional test, like this: :)
```R
if(title_position == "topleft" && title_width > legend_width) {
```
It tests `title_position` first, so I can avoid the second part with `>` error by setting `title_position="topcenter"`. I know this is not a real solution, but it is very low risk that does not change any functional R code.

I appreciate your help!

**Comment by jokergoo** [comment]
This problem is due to the upgrade of grid package from R3.* to R4.\*. I tried to make ComplexHeatmap compatible with R <= 3.\*, but since then I am mainly working on R4.\*, there must be something I missed for R3.*. 

I will have a look.

**Comment by jokergoo** [comment]
I found the reason. The expression e.g. `-unit(1, "mm")` is not allowed with grid3 while it is valid with grid4. Now I have changed it to `-1*unit(1, "mm")` and now the error should go away.

Please update from GitHub.

**Comment by jmw86069** [comment]
Confirmed, and thank you so much! It even works for all the different `title_position` values. (Bonus points!)

My test code fwiw:
```R
col_fun <- circlize::colorRamp2(
   breaks=-2:2,
   colors=c("navy", "dodgerblue", "white", "tomato", "red3"))
lgd <- ComplexHeatmap::Legend(col_fun=col_fun,
   title="log2FC",
   title_position="topleft",
   direction="horizontal")
plot(0, 0, type="n", axes=FALSE)
ComplexHeatmap::draw(lgd)
```

Meanwhile, I learned about adding custom legends beside heatmaps in the `draw()` function, that's really nice! (For example, it might be useful for me to build color legends to inspect/confirm before rendering extremely large heatmaps.) Your code is fun to use.

Closing this issue, and it is fully resolved. Thank you again!
